dbtools/pg.py

```diff
@@ -677,18 +686,13 @@ def get_table_count(self, table, schema=None):
 
         return count
 
-    def get_table_columns(self, table, schema=None) -> List:
-        """Get columns in passed table."""
-        if schema is not None:
-            self.cursor.execute(
-                sql.SQL("SELECT * FROM {}.{} LIMIT 0").format(
-                    sql.Identifier(schema), sql.Identifier(table)
-                )
-            )
-        else:
-            self.cursor.execute(sql.SQL("SELECT * FROM {} LIMIT 0").format(sql.Identifier(table)))
-
-        columns = [d[0] for d in self.cursor.description]
+    def get_table_columns(self, table: str, schema: str) -> List[str]:
+        columns_sql = sql.SQL(
+            "SELECT column_name FROM information_schema.columns "
+            "WHERE table_schema = {} AND table_name = {};"
+        ).format(sql.Literal(schema), sql.Literal(table))

```

Comment:
I don't have a great handle on this (just poking through [these docs](https://www.psycopg.org/docs/sql.html#psycopg2.sql.Identifier))  but it seems like keeping `sql.Identifier()` over `sql.Literal()` may be preferable here?

Reply:
Basically those two functions(?) just add the appropriate quotes to the query. In this case, we are actually wanting the literal name of the table since it's actually being used as a value in a column in the `information_schema.columns` table, not as an actual reference to the table itself.